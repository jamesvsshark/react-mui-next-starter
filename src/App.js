import React from "react";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import Button from "@material-ui/core/Button";
import Dialog from "@material-ui/core/Dialog";
import DialogTitle from "@material-ui/core/DialogTitle";
import DialogContent from "@material-ui/core/DialogContent";
import DialogContentText from "@material-ui/core/DialogContentText";
import DialogActions from "@material-ui/core/DialogActions";
import Typography from "@material-ui/core/Typography";
import { withStyles } from "@material-ui/core/styles";

class App extends React.Component {
  handleClose = () => {
    this.props.toggleDialogue();
  };

  handleClick = () => {
    this.props.toggleDialogue();
  };

  render() {
    const { classes, sample: state } = this.props;

    return (
      <div className={classes.root}>
        <Typography variant="display1" gutterBottom>
          React-Redux-MUI-Starter
        </Typography>
        <Typography variant="subheading" gutterBottom>
          Happy coding
        </Typography>
        <br />
        <Dialog open={state.openDialogue} onClose={this.handleClose}>
          <DialogTitle>Super Secret Password</DialogTitle>
          <DialogContent>
            <DialogContentText>1-2-3-4-5</DialogContentText>
          </DialogContent>
          <DialogActions>
            <Button color="primary" onClick={this.handleClose}>
              OK
            </Button>
          </DialogActions>
        </Dialog>

        <Button variant="raised" color="secondary" onClick={this.handleClick}>
          Super Secret Password
        </Button>
      </div>
    );
  }
}

App.propTypes = {
  classes: PropTypes.object.isRequired,
  todos: PropTypes.array,
  toggleTodo: PropTypes.func
};

const styles = theme => ({
  root: {
    textAlign: "center",
    padding: theme.spacing.unit * 10
  }
});

const mapStateToProps = state => ({
  sample: state.sample
});

const mapDispatchToProps = dispatch => ({
  toggleDialogue: todo => {
    dispatch({ type: "TOGGLE_DIALOGUE" });
  }
});

export default connect(mapStateToProps, mapDispatchToProps)(
  withStyles(styles)(App)
);
