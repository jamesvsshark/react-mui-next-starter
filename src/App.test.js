import React from "react";
import ReactDOM from "react-dom";
import { App } from "./App";

it("renders without crashing", () => {
  const div = document.createElement("div");
  const props = { classes: {}, sample: {}, toggleDialogue: () => {} };
  ReactDOM.render(<App {...props} />, div);
  ReactDOM.unmountComponentAtNode(div);
});
