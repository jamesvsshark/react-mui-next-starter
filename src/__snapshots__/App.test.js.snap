// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main app should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    classes={Object {}}
    sample={Object {}}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <WithStyles(Typography)
          gutterBottom={true}
          variant="display1"
>
          React-Redux-MUI-Starter
</WithStyles(Typography)>,
        <WithStyles(Typography)
          gutterBottom={true}
          variant="subheading"
>
          Happy coding
</WithStyles(Typography)>,
        <br />,
        <WithStyles(Dialog)
          onClose={[Function]}
          open={undefined}
>
          <WithStyles(DialogTitle)>
                    Super Secret Password
          </WithStyles(DialogTitle)>
          <WithStyles(DialogContent)>
                    <WithStyles(DialogContentText)>
                              1-2-3-4-5
                    </WithStyles(DialogContentText)>
          </WithStyles(DialogContent)>
          <WithStyles(DialogActions)>
                    <WithStyles(Button)
                              color="primary"
                              onClick={[Function]}
                              variant="outlined"
                    >
                              OK
                    </WithStyles(Button)>
          </WithStyles(DialogActions)>
</WithStyles(Dialog)>,
        <WithStyles(Button)
          color="primary"
          onClick={[Function]}
          size="large"
          variant="outlined"
>
          Super Secret Password
</WithStyles(Button)>,
      ],
      "className": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "React-Redux-MUI-Starter",
          "gutterBottom": true,
          "variant": "display1",
        },
        "ref": null,
        "rendered": "React-Redux-MUI-Starter",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Happy coding",
          "gutterBottom": true,
          "variant": "subheading",
        },
        "ref": null,
        "rendered": "Happy coding",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <WithStyles(DialogTitle)>
              Super Secret Password
</WithStyles(DialogTitle)>,
            <WithStyles(DialogContent)>
              <WithStyles(DialogContentText)>
                            1-2-3-4-5
              </WithStyles(DialogContentText)>
</WithStyles(DialogContent)>,
            <WithStyles(DialogActions)>
              <WithStyles(Button)
                            color="primary"
                            onClick={[Function]}
                            variant="outlined"
              >
                            OK
              </WithStyles(Button)>
</WithStyles(DialogActions)>,
          ],
          "onClose": [Function],
          "open": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Super Secret Password",
            },
            "ref": null,
            "rendered": "Super Secret Password",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <WithStyles(DialogContentText)>
                1-2-3-4-5
</WithStyles(DialogContentText)>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "1-2-3-4-5",
              },
              "ref": null,
              "rendered": "1-2-3-4-5",
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <WithStyles(Button)
                color="primary"
                onClick={[Function]}
                variant="outlined"
>
                OK
</WithStyles(Button)>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "OK",
                "color": "primary",
                "onClick": [Function],
                "variant": "outlined",
              },
              "ref": null,
              "rendered": "OK",
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Super Secret Password",
          "color": "primary",
          "onClick": [Function],
          "size": "large",
          "variant": "outlined",
        },
        "ref": null,
        "rendered": "Super Secret Password",
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <WithStyles(Typography)
            gutterBottom={true}
            variant="display1"
>
            React-Redux-MUI-Starter
</WithStyles(Typography)>,
          <WithStyles(Typography)
            gutterBottom={true}
            variant="subheading"
>
            Happy coding
</WithStyles(Typography)>,
          <br />,
          <WithStyles(Dialog)
            onClose={[Function]}
            open={undefined}
>
            <WithStyles(DialogTitle)>
                        Super Secret Password
            </WithStyles(DialogTitle)>
            <WithStyles(DialogContent)>
                        <WithStyles(DialogContentText)>
                                    1-2-3-4-5
                        </WithStyles(DialogContentText)>
            </WithStyles(DialogContent)>
            <WithStyles(DialogActions)>
                        <WithStyles(Button)
                                    color="primary"
                                    onClick={[Function]}
                                    variant="outlined"
                        >
                                    OK
                        </WithStyles(Button)>
            </WithStyles(DialogActions)>
</WithStyles(Dialog)>,
          <WithStyles(Button)
            color="primary"
            onClick={[Function]}
            size="large"
            variant="outlined"
>
            Super Secret Password
</WithStyles(Button)>,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "React-Redux-MUI-Starter",
            "gutterBottom": true,
            "variant": "display1",
          },
          "ref": null,
          "rendered": "React-Redux-MUI-Starter",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Happy coding",
            "gutterBottom": true,
            "variant": "subheading",
          },
          "ref": null,
          "rendered": "Happy coding",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <WithStyles(DialogTitle)>
                Super Secret Password
</WithStyles(DialogTitle)>,
              <WithStyles(DialogContent)>
                <WithStyles(DialogContentText)>
                                1-2-3-4-5
                </WithStyles(DialogContentText)>
</WithStyles(DialogContent)>,
              <WithStyles(DialogActions)>
                <WithStyles(Button)
                                color="primary"
                                onClick={[Function]}
                                variant="outlined"
                >
                                OK
                </WithStyles(Button)>
</WithStyles(DialogActions)>,
            ],
            "onClose": [Function],
            "open": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Super Secret Password",
              },
              "ref": null,
              "rendered": "Super Secret Password",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <WithStyles(DialogContentText)>
                  1-2-3-4-5
</WithStyles(DialogContentText)>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "1-2-3-4-5",
                },
                "ref": null,
                "rendered": "1-2-3-4-5",
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <WithStyles(Button)
                  color="primary"
                  onClick={[Function]}
                  variant="outlined"
>
                  OK
</WithStyles(Button)>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "OK",
                  "color": "primary",
                  "onClick": [Function],
                  "variant": "outlined",
                },
                "ref": null,
                "rendered": "OK",
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Super Secret Password",
            "color": "primary",
            "onClick": [Function],
            "size": "large",
            "variant": "outlined",
          },
          "ref": null,
          "rendered": "Super Secret Password",
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
